### PR TITLE
added full text description in config

### LINKED ISSUE #5: Incomplete Output Functionality
The emulator requires the input file to include a value for idtl, which determines the output level. Currently, only 0 (standard) and 2 (full) are supported. Level 5 (full with text descriptions) is not implemented.

### LINKED ISSUE #53: Implement dependent based output when they are provided
Implement dependent based output when they are provided in in the input csv

### PR DESCRIPTION
**Pull request by noman404**
added full text description in config
Resolves #5 #61 #62

### CODE REVIEW

File: policyengine_taxsim/config/variable_mappings.yaml [hunk 1]
```diff
@@ -38,250 +54,412 @@ policyengine_to_taxsim:
...
     variable: income_tax_before_refundable_credits
     implemented: true
     idtl:
      - full: 2
      - full_text: 5
+    text_description: "28. Income Tax Before Credits \n Total Credits"
+    group_order: 3
+    full_text_group: "Federal Tax Calculation"
+    group_column: 2
   v29:
```
Comment (PavelMakarchuk) [inline policyengine_taxsim/config/variable_mappings.yaml:294]: Same as V17:
should be 2 outputs:

We were able to generate an output previously - not sure which variable it was - will reach out to Martin

`taxsim_tfica` - this is the taxpayer share

File: policyengine_taxsim/config/variable_mappings.yaml [hunk 2]
```diff
@@ -38,250 +54,412 @@ policyengine_to_taxsim:
...
     implemented: false
     idtl:
      - full: 2
      - full_text: 5
+    text_description: "32. AGI"
+    group_order: 4
+    full_text_group: "State Tax Calculation"
+    group_column: 2
   v33:
     variable: not_implemented_in_pe
```
Comment (PavelMakarchuk) [inline policyengine_taxsim/config/variable_mappings.yaml:335]: We have state exemptions - just not in one variable (naming varies) - is this a change in the US package that needs to happen?

Comment (PavelMakarchuk) [reply policyengine_taxsim/config/variable_mappings.yaml:335]: generally using 'state_exemptions' should produce a valid result in ~80% of the states

File: policyengine_taxsim/config/variable_mappings.yaml [hunk 3]
```diff
@@ -38,250 +54,412 @@ policyengine_to_taxsim:
...
     implemented: false
     idtl:
      - full: 2
      - full_text: 5
+    text_description: "37. Property Tax Credit"
+    group_order: 4
+    full_text_group: "State Tax Calculation"
+    group_column: 2
   v38:
     variable: not_implemented_in_pe
```
Comment (PavelMakarchuk) [inline policyengine_taxsim/config/variable_mappings.yaml:385]: we can generally use 'state_cdcc' with some exemptions

File: policyengine_taxsim/config/variable_mappings.yaml [hunk 4]
```diff
@@ -38,250 +54,412 @@ policyengine_to_taxsim:
...
+    group_order: 4
+    full_text_group: "State Tax Calculation"
+    group_column: 2
   v39:
     variable: not_implemented_in_pe
     implemented: false
     idtl:
      - full: 2
      - full_text: 5
+    text_description: "39. EIC \n Energy|Fuel Credit \n Child Tax Credit"
```
Comment (PavelMakarchuk) [inline policyengine_taxsim/config/variable_mappings.yaml:400]: EIC: 'state_eitc' - same limitations as cdcc

Child Tax credit: 'state_ctc' same limitations as cdcc

File: policyengine_taxsim/config/variable_mappings.yaml [hunk 5]
```diff
@@ -38,250 +54,412 @@ policyengine_to_taxsim:
...
+    group_order: 4
+    full_text_group: "State Tax Calculation"
+    group_column: 2
   v40:
     variable: not_implemented_in_pe
     implemented: false
     idtl:
      - full: 2
      - full_text: 5
+    text_description: "40. Total Credits"
```
Comment (PavelMakarchuk) [inline policyengine_taxsim/config/variable_mappings.yaml:410]: this should be a sum of
'state_non_refundable_credit'
and 
'state_refundable_credits'

File: policyengine_taxsim/config/variable_mappings.yaml [hunk 6]
```diff
@@ -38,250 +54,412 @@ policyengine_to_taxsim:
...
     implemented: false
     idtl:
      - full: 2
      - full_text: 5
+    text_description: "41. Bracket Rate \n  State Tax after Credits"
+    group_order: 4
+    full_text_group: "State Tax Calculation"
+    group_column: 2
   v42:
     variable: self_employment_income
```
Comment (PavelMakarchuk) [inline policyengine_taxsim/config/variable_mappings.yaml:425]: this should be the `qualified_business_income_deduction`

File: policyengine_taxsim/config/variable_mappings.yaml [hunk 7]
```diff
@@ -38,250 +54,412 @@ policyengine_to_taxsim:
...
+    group_order: 4
+    full_text_group: "State Tax Calculation"
+    group_column: 2
   v43:
     variable: net_investment_income_tax
     implemented: true
     idtl:
      - full: 2
      - full_text: 5
+    text_description: "Medicare tax on Unearned Income"
```
Comment (PavelMakarchuk) [inline policyengine_taxsim/config/variable_mappings.yaml:440]: v44 is medicare right?

Comment (PavelMakarchuk) [reply policyengine_taxsim/config/variable_mappings.yaml:440]: this is net investment income tax

File: policyengine_taxsim/config/variable_mappings.yaml [hunk 8]
```diff
@@ -38,250 +54,412 @@ policyengine_to_taxsim:
...
+    group_order: 5
+    full_text_group: "Regular Income Tax"
+    group_column: 1
   v44:
     variable: employee_medicare_tax
     implemented: true
     idtl:
      - full: 2
-     - full_text: 5
+    text_description: "Not available in taxsim full text description"
```
Comment (PavelMakarchuk) [inline policyengine_taxsim/config/variable_mappings.yaml:449]: medicare

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]
Some comments on the inputs - lmk if those make sense

**Comment by MaxGhenis** [comment]
what's the status on this pr?

File: policyengine_taxsim/config/variable_mappings.yaml [hunk 9]
```diff
@@ -38,250 +54,703 @@ policyengine_to_taxsim:
...
+    variable: multiple_variables
     implemented: true
     idtl:
      - full: 2
      - full_text: 5
+    text_description: "27. AMT"
+    group_order: 3
+    full_text_group: "Federal Tax Calculation"
+    group_column: 2
+    variables:
```
[the remark this answers is not in the document]
Comment (noman404) [reply policyengine_taxsim/config/variable_mappings.yaml]: updated

File: policyengine_taxsim/config/variable_mappings.yaml [hunk 10]
```diff
@@ -302,5 +771,101 @@ taxsim_to_policyengine:
...
+    additional_income_units:
+      - taxable_interest_income:
+          - intrec
+      - rent:
+          - rentpaid
+      - qualified_dividend_income:
+          - dividends
+      - long_term_capital_gains:
+          - ltcg
+      - self_employment_income:
```
[the remark this answers is not in the document]
Comment (noman404) [reply policyengine_taxsim/config/variable_mappings.yaml:848]: planning this part to restructure after confirming with dan for the other possible input mapping, currently there are some variables which does not have any match, like these 

```
      - rental_income:
      - w2_wages_from_qualified_business:
      - business_is_sstb:
      - business_is_qualified:
      - social_security:
```

[the remark this answers is not in the document]
Comment (noman404) [reply policyengine_taxsim/config/variable_mappings.yaml:848]: comment from Dan

interest income for 
 - rental_income:
 - w2_wages_from_qualified_business:
 - business_is_sstb:
 - business_is_qualified
 
and gssi for
 - social_security:

**Comment by MaxGhenis** [comment]
Merging as we have issues for remaining tasks

**Pull request merged**